src/ccnl-core/src/ccnl-malloc.c

```diff
@@ -64,10 +88,17 @@ void*
 debug_calloc(size_t n, size_t s, const char *fn, int lno, char *tstamp)
 #endif
 {
-    /** TODO: potential integer overflow by n * z operation */
-    void *p = debug_malloc(n * s, fn, lno, tstamp);
-    if (p)
-        memset(p, 0, n*s);
+    size_t size;
+    void *p = NULL;
+    /** can the operation be performed without causing an integer overflow */
+    if (!INT_MULT_OVERFLOW(n, s, size)) {
```

Comment:
Here I get another error:
 result argument to overflow builtin must be a pointer to a non-const integer.
Same error in line 130 and line 176

Reply:
Fixed. The third argument has to be a pointer.